lang/csharp/src/apache/test/Reflect/TestArray.cs

```diff
@@ -0,0 +1,243 @@
+        {
+            ""namespace"": ""MessageTypes"",
+            ""type"": ""array"",
+            ""helper"": ""arrayOfA"",
+            ""items"": {
+                ""type"": ""record"",
+                ""doc"": ""A simple type with a fixed."",
+                ""name"": ""A"",
+                ""fields"": [
+                    { ""name"" : ""S"", ""type"" : ""string"" }
+                ]
+            }
+        }";
+
+
+
+        [TestCase]
+        public void ListTest()
+        {
+            var schema = Schema.Parse(_simpleList);
+            var fixedRecWrite = new List<string>() {"value"};
+
+            var writer = new ReflectWriter<List<string>>(schema);
+            var reader = new ReflectReader<List<string>>(schema, schema);
+
+            using (var stream = new MemoryStream(256))
+            {
+                writer.Write(fixedRecWrite, new BinaryEncoder(stream));
+                stream.Seek(0, SeekOrigin.Begin);
+                var fixedRecRead = reader.Read(new BinaryDecoder(stream));
+                Assert.IsTrue(fixedRecRead.Count == 1);
+                Assert.AreEqual(fixedRecWrite[0],fixedRecRead[0]);
+            }
+        }
+
+        [TestCase]
+        public void ListRecTest()
+        {
+            var schema = Schema.Parse(_recordList);
+            var fixedRecWrite = new List<ListRec>() { new ListRec() { S = "hello"}};
+
+            var writer = new ReflectWriter<List<ListRec>>(schema);
+            var reader = new ReflectReader<List<ListRec>>(schema, schema);
+
+            using (var stream = new MemoryStream(256))
+            {
+                writer.Write(fixedRecWrite, new BinaryEncoder(stream));
+                stream.Seek(0, SeekOrigin.Begin);
+                var fixedRecRead = reader.Read(new BinaryDecoder(stream));
+                Assert.IsTrue(fixedRecRead.Count == 1);
+                Assert.AreEqual(fixedRecWrite[0].S,fixedRecRead[0].S);
+            }
+        }
+
+        public class ConcurrentQueueHelper<T> : ArrayHelper
+        {
+
+            /// <summary>
+            /// Return the number of elements in the array.
+            /// </summary>
+            /// <value></value>
+            public override int Count()
+            {
+                ConcurrentQueue<T> e = (ConcurrentQueue<T>)Enumerable;
+                return e.Count;
+            }
+            /// <summary>
+            /// Add an element to the array.
+            /// </summary>
+            /// <value></value>
+            public override void Add(object o)
+            {
+                ConcurrentQueue<T> e = (ConcurrentQueue<T>)Enumerable;
+                e.Enqueue((T)o);
+            }
+            /// <summary>
+            /// Clear the array.
+            /// </summary>
+            /// <value></value>
+            public override void Clear()
+            {
+                ConcurrentQueue<T> e = (ConcurrentQueue<T>)Enumerable;
+                e.Clear();
```

Comment:
This unit test project runs agains NET461 on Windows. NET461 does not define `ConcurrentQueue{T}.Clear()`. Instead, for net461, we should dequeue all items.

```suggestion
#if NET461
                while (e.TryDequeue(out _)) { }
#else
                e.Clear();
#endif
```